Optimize fullscreen prompt handling

wavexx opened the pull request:
Modify ``lcd_show_multiscreen_message_two_choices_and_wait_P`` to also handle single-screen or empty (no-clear) prompts, making other functions redundant.

Saves 76 bytes.

This is based on top of PR #3089 (there's just one additional commit). It was split off because it changes the layout of existing Yes/No of messages: the previous prompt would use the last two lines of the LCD, while the new prompt is using just the last line of the LCD instead.

Before:
![2021-03-28T175413](https://user-images.githubusercontent.com/1017726/112759261-ba3e7100-8fe1-11eb-9eb2-e974598169ee.png)

After:
![2021-03-28T175118](https://user-images.githubusercontent.com/1017726/112759263-be6a8e80-8fe1-11eb-91a6-6b4e968e35af.png)

Translation do not require updates, since the Yes/No translation was already the same in both implementations and thus should always fit already.

wavexx committed:
Unify fullscreen/multiscreen/yes-no prompt message handling

Modify lcd_show_multiscreen_message_two_choices_and_wait_P to also
handle single-screen or empty (no-clear) prompts, making other functions
redundant. Saves 76 bytes.

Change existing functions to simply call
lcd_show_multiscreen_message_two_choices_and_wait_P with the correct
arguments.

This changes the prompt of existing Yes/No messages: the previous prompt
would use the last two lines of the LCD, while the new prompt is using
just the last line of the LCD instead.

Translation do not require updates, since the Yes/No translation was
already the same in both implementations.

3d-gussner commented on Firmware/ultralcd.cpp:
```diff
@@ -3472,8 +3472,21 @@ int8_t lcd_show_multiscreen_message_yes_no_and_wait_P(const char *msg, bool allo
+    lcd_set_cursor(0, 3);
+    lcd_print(first_selected? '>': ' ');
+    lcd_puts_P(first_choice);
+    lcd_set_cursor(7, 3);
```
Can you change it to `lcd_set_cursor(10, 3);` so the `Yes/No` both translations can be up to 8 chars?

wavexx replied:
We need to ensure that all the possible translation for "choice 2" (not just no - but all the wizard stuff) fits into 8 characters though.

Otherwise that's pretty trivial

wavexx replied:
Done!

wavexx committed:
Allow choice positioning. Balance Yes/No messages.

Add an additional parameter to control the position of second choice
prompt position (while defaulting to the old).

This allows Yes/No prompts to be equally spaced.

wavexx commented:
We now allow to position the second prompt on an arbitrary column, allowing choices to be flexible in spacing.
As a a side-effect, we can now balance the space of Yes/No prompts:

![2021-04-05T212107](https://user-images.githubusercontent.com/1017726/113617023-679a3000-9645-11eb-8158-408b166770cd.png)

sketch size didn't seem to change, so we just got a free feture? ;)

3d-gussner commented:
Reminder after it got merged:
- Review translations to un-shorten these again to make it more readable.

3d-gussner commented:
@wavexx I have found few other messages that could be improved:
- Stop SD card print
- Selftest FANs
So if you like please change it as well.

```
void lcd_sdcard_stop()
{

	lcd_puts_at_P(0, 0, _T(MSG_STOP_PRINT));
	lcd_puts_at_P(2, 2, _T(MSG_NO));
	lcd_puts_at_P(2, 3, _T(MSG_YES));
	lcd_putc_at(0, 2, ' ');
	lcd_putc_at(0, 3, ' ');

....
	lcd_puts_at_P(1, 2, _T(MSG_SELFTEST_FAN_YES));
	lcd_putc_at(0, 3, '>');
	lcd_puts_at_P(1, 3, _T(MSG_SELFTEST_FAN_NO));
....
```

3d-gussner commented on Firmware/messages.c:
```diff
@@ -26,7 +26,7 @@ const char MSG_COMMUNITY_MADE[] PROGMEM_I1 = ISTR("Community made"); ////c=18
 const char MSG_COOLDOWN[] PROGMEM_I1 = ISTR("Cooldown"); ////c=18
 const char MSG_CRASH[] PROGMEM_I1 = ISTR("Crash"); ////c=7
-const char MSG_CRASH_DETECTED[] PROGMEM_I1 = ISTR("Crash detected."); ////c=20
+const char MSG_CRASH_DETECTED[] PROGMEM_I1 = ISTR("Crash detected."); ////c=17 r=1
```
Why c=17?

wavexx replied:
This is because "XY " can/is added at the beginning of the string now, so we have a little less space.

3d-gussner commented:
I was thinking if we can improve changed correctly as well.
```
01234567890123456789
Changed correctly and
Color is correct? 
>Yess>Nooo>Extr. more
```
We have a limit of c=4 for MSG_YES and MSG_NO which means we could have triple choice with c=9
<space or `>`:1><`YES`:4><space or `>`:1><`NO`:4><space or `>`:1><third option>
20-1-4-1-4-1=9

3d-gussner commented:
Tried to crash Y axis and the output is odd
![image](https://user-images.githubusercontent.com/25530011/155120199-caf420d3-c70c-4a53-b571-9cff5c451246.png)

leptun commented:
@3d-gussner I think the branch is outdated. I already fixed that bug. See https://github.com/prusa3d/Prusa-Firmware/pull/3415

3d-gussner reviewed (approved):
Works and looks fine. Approved

Content approved the layout changes and @gudnimg is building on this one for the error messags.
